### ISSUE TITLE
Designing base code for feature extraction

### ISSUE THREAD

**Issue by keighrim** [keighrim/bananaNER#4]
### Need a class that has at least:
1. **read()** - a method to load up train.gold file into a list of sentences (or any other data structure)
2. any necessary method that return tokens, sentences, <token, postag> pairs at token level, <token, postag> pairs at sentence level 
3. **feature_functions = []** - a list of names of functions, doing one job for each. Since we already have POS tags, something like pos_tag() can be the first element of this list
4. **extract()** - method to traverse all functions in feature_functions to create feature vector
### feature vector data structure

Basically it should be a table-like structure, where each column represents each feat_func in feat_funcs, and each row is for each token

|  | fn1 | fn2 | fn3 | ... |
| --- | --- | --- | --- | --- |
| token1 | val1(1) | val2(1) | val3(1) | ... |
| token2 | val1(2) | val2(2) | val3(2) | ... |
| token3 | val1(3) | val2(3) | val3(3) | ... |
| ... | ... | ... | ... | ... |

Any additions or suggestions are welcomed.
